cmd/commands/init.go

```diff
@@ -0,0 +1,122 @@
+package commands
+
+import (
+	"io/ioutil"
+	"path"
+
+	"github.com/urfave/cli"
+
+	cmn "github.com/tendermint/go-common"
+)
+
+var InitCmd = cli.Command{
+	Name:      "init",
+	Usage:     "Initialize a basecoin blockchain",
+	ArgsUsage: "",
+	Action: func(c *cli.Context) error {
+		return cmdInit(c)
+	},
+	Flags: []cli.Flag{
+		ChainIDFlag,
+	},
+}
+
+func cmdInit(c *cli.Context) error {
+	rootDir := BasecoinRoot("")
+
+	cmn.EnsureDir(rootDir, 0777)
+
+	// initalize basecoin
+	genesisFile := path.Join(rootDir, "genesis.json")
+	privValFile := path.Join(rootDir, "priv_validator.json")
+	key1File := path.Join(rootDir, "key.json")
+	key2File := path.Join(rootDir, "key2.json")
+
+	if err := ioutil.WriteFile(genesisFile, []byte(genesisJSON), 0644); err != nil {
```

Comment:
For readability, during refactor to cobra, error returns are not-necessarily needed can just exit. As so, we can condense code further using pkg/errors (for tracing) and a function variable, will implement during refactor
```golang
exitOnErr := func(err error){
     if err != nil {
          cmn.Exit(fmt.Sprintf("%+v\n", err))
     }
}

exitOnErr(ioutil.WriteFile(genesisFile, []byte(genesisJSON), 0644))
... 
```